server/src/main/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorFieldMapper.java

```diff
@@ -0,0 +1,443 @@
+            Integer dims,
+            IndexVersion indexCreatedVersion,
+            Map<String, String> meta
+        ) {
+            super(name, false, false, true, TextSearchInfo.NONE, meta);
+            this.elementType = elementType;
+            this.dims = dims;
+            this.indexCreatedVersion = indexCreatedVersion;
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
+            if (format != null) {
+                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
+            }
+            return new ArraySourceValueFetcher(name(), context) {
+                @Override
+                protected Object parseSourceValue(Object value) {
+                    return value;
+                }
+            };
+        }
+
+        @Override
+        public DocValueFormat docValueFormat(String format, ZoneId timeZone) {
+            throw new IllegalArgumentException(
+                "Field [" + name() + "] of type [" + typeName() + "] doesn't support docvalue_fields or aggregations"
+            );
+        }
+
+        @Override
+        public boolean isAggregatable() {
+            return false;
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+            return new MultiVectorIndexFieldData.Builder(name(), CoreValuesSourceType.KEYWORD, indexCreatedVersion, dims, elementType);
+        }
+
+        @Override
+        public Query existsQuery(SearchExecutionContext context) {
+            return new FieldExistsQuery(name());
+        }
+
+        @Override
+        public Query termQuery(Object value, SearchExecutionContext context) {
+            throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support term queries");
+        }
+
+        int getVectorDimensions() {
+            return dims;
+        }
+
+        DenseVectorFieldMapper.ElementType getElementType() {
+            return elementType;
+        }
+    }
+
+    private final IndexVersion indexCreatedVersion;
+
+    private MultiDenseVectorFieldMapper(
+        String simpleName,
+        MappedFieldType fieldType,
+        BuilderParams params,
+        IndexVersion indexCreatedVersion
+    ) {
+        super(simpleName, fieldType, params);
+        this.indexCreatedVersion = indexCreatedVersion;
+    }
+
+    @Override
+    public MultiDenseVectorFieldType fieldType() {
+        return (MultiDenseVectorFieldType) super.fieldType();
+    }
+
+    @Override
+    public boolean parsesArrayValue() {
+        return true;
+    }
+
+    @Override
+    public void parse(DocumentParserContext context) throws IOException {
+        if (context.doc().getByKey(fieldType().name()) != null) {
```

Comment:
Should this be deleted, as this field type supports multiple values

Reply:
I suppose, it supports a single array of arrays where all the arrays have the same dimension. I wonder if its even possible to cause this path to throw...

Reply:
I added a bunch of testing, this doesn't seem to throw, but for paranoia, I would like to keep this check if thats ok.